### PR TITLE
Move the last calls in NativeWindow to Interop

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Fix a few Libraries references. Add some safe helpers to `PARAM`.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1780)

### CODE REVIEW

File: src/Common/src/Interop/User32/Interop.CS.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        /// <summary>
+        ///  Class styles
+        /// </summary>
+        public enum CS : uint
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.CS.cs]: In the spirit and in line with other refactors:
```suggestion
        public enum ClassStyle : uint
```

Looking at the use, it probably should be decorated as `Flags`, and then name `ClassStyles`

Also there are more members, perhaps it is worth adding them all https://github.com/tpn/winsdk-10/blob/master/Include/10.0.14393.0/um/WinUser.h#L2856-L2874

File: src/Common/src/Interop/User32/Interop.WS.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        /// <summary>
+        ///  Window styles
+        /// </summary>
+        public enum WS : uint
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.WS.cs]:
```suggestion
        public enum WindowStyle : uint
```

Looking at the use, it probably should be decorated as `Flags`, and then name `WindowStyles`

**Comment by RussKie** [comment]
Please fix the MC.

File: src/Common/src/Interop/User32/Interop.CS.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/User32/Interop.CS.cs]: As per https://github.com/dotnet/winforms/pull/1742#discussion_r321841577 I don't think that is the right choice.

For diagnostic purposes I think having the additional defines are useful, but they do bloat the size of the assembly, so we should be thinking about the value in each case and not doing it blindly. One option is to comment out additional defines when we only use a few.

Comment (JeremyKuhne) [reply src/Common/src/Interop/User32/Interop.CS.cs]: I've changed a few of the existing ones to this pattern. Here is an example of where it really starts making a difference:

https://github.com/dotnet/winforms/pull/1780/commits/0abd60a6aa3f062171a96158ca0dd5e66ccb55df#diff-c6a8af157a9eb8becc613c8d53ce2c1fL622

Comment (RussKie) [reply src/Common/src/Interop/User32/Interop.CS.cs]: I can't say I find it particularly exciting, but at least it is consistent. 👍

File: src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs [hunk 3]
```diff
@@ -524,13 +524,14 @@ public virtual void DestroyHandle()
             {
                 if (Handle != IntPtr.Zero)
                 {
-                    if (!UnsafeNativeMethods.DestroyWindow(new HandleRef(this, Handle)))
+                    if (User32.DestroyWindow(this).IsFalse())
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: We have about 25 uses of `== BOOL.FALSE` and `!= BOOL.FALSE`. Should we replace them too?

File: src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs [hunk 4]
```diff
@@ -338,9 +338,9 @@ internal void AssignHandle(IntPtr handle, bool assignUniqueID)
                 // This shouldn't be possible.
                 Debug.Assert(_priorWindowProcHandle != _windowProcHandle, "Uh oh! Subclassed ourselves!!!");
 
-                if (assignUniqueID &&
-                    (unchecked((int)((long)User32.GetWindowLong(this, User32.GWL.STYLE))) & NativeMethods.WS_CHILD) != 0 &&
-                     unchecked((int)((long)User32.GetWindowLong(this, User32.GWL.ID))) == 0)
+                if (assignUniqueID
+                    && ((User32.WS)PARAM.ToUInt(User32.GetWindowLong(this, User32.GWL.STYLE))).HasFlag(User32.WS.CHILD)
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: Yes it has.

File: src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs (hunk 3 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: Yeah, I'll go through and clean them.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: @hughbe can you please align the changes in your PRs as well.

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: @hughbe can you please align the changes in your PRs as well.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: Yeah I’ll rebase after this is merged !!!

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.cs]: I'll do that in a separate PR.

**Pull request merged**